src/idiomatic/leveraging-the-type-system/extension-traits/extending-other-traits.md

```diff
@@ -0,0 +1,95 @@
+---
+minutes: 15
+---
+
+# Extending Other Traits
+
+As with types, it may be desirable to **extend foreign traits**. In particular,
+to attach new methods to _all_ implementors of a given trait.
+
+```rust
+mod ext {
+    use std::fmt::Display;
+
+    pub trait DisplayExt {
+        fn quoted(&self) -> String;
+    }
+
+    impl<T: Display> DisplayExt for T {
+        fn quoted(&self) -> String {
+            format!("'{}'", self)
+        }
+    }
+}
+
+pub use ext::DisplayExt as _;
+
+assert_eq!("dad".quoted(), "'dad'");
+assert_eq!(4.quoted(), "'4'");
+assert_eq!(true.quoted(), "'true'");
+```
+
+<details>
+
+- Highlight how we added new behaviour to _multiple_ distinct types at once.
+  `.quoted()` can be called on string slices, numbers and booleans since they
+  all implement the `Display` trait.
+
+  This flavour of the extension trait pattern is built on top of
+  [_blanket implementations_][1].
+
+  Blanket implementations allow us to implement a trait for a generic type `T`,
+  as long as it satisfies the trait bounds specified in the `impl` block. In
+  this case, the only requirement is that `T` implements the `Display` trait.
+
+- Draw the students attention to the implementation of `DisplayExt::quoted`: we
+  can't make any assumptions about the type of `T` other than that it implements
+  `Display`. All our logic must either use methods from `Display` or
+  functions/macros that doesn't require `T` to implement any other trait.
+
+  We could introduce additional trait bounds on `T`, but it would restrict the
+  set of types that can leverage the extension trait.
+
+- Conventionally, the extension trait is named after the trait it extends,
+  following by the `Ext` suffix. In the example above, `DisplayExt`.
```

Comment:
I'm not sure I agree with this, I think personally I'd name the trait something like `Quote` or `Quoted`. Calling it `DisplayExt` is reasonable if this is the only trait extending `Display`, but what if you had multiple traits extending `Display` in different ways? For example, I might have another trait that has a blanket impl on `T: Display + Clone` or something. Calling that `DisplayCloneExt` would be cumbersome and doesn't describe what behavior it's extending those traits with.